### PR TITLE
DB-6008: suppress false constraint violation during retry

### PR DESCRIPTION
**Pull request by jyuanca**

### CODE REVIEW

**Comment by dgomezferro** [comment]
Would it be possible to add some tests? Or is it too hard to reproduce?

**Comment by jyuanca** [comment]
@dgomezferro This is hard to reproduce.

File: pipeline_api/src/main/java/com/splicemachine/pipeline/constraint/UniqueConstraintChecker.java [hunk 1]
```diff
@@ -48,7 +49,18 @@ public UniqueConstraintChecker(boolean isPrimaryKey,
     @Override
     public MutationStatus checkConstraint(KVPair mutation, DataResult existingRow) throws IOException {
         // There is an existing row, if this is an insert then fail.
```
Comment (dgomezferro) [inline pipeline_api/src/main/java/com/splicemachine/pipeline/constraint/UniqueConstraintChecker.java]: Update this comment with something similar to what's in UniqueConstraint.java please

**Comment by jyuanca** [comment]
@dgomezferro added comments.

**Pull request merged**